src/src/hooks/Error.js

```diff
@@ -0,0 +1,48 @@
+import { useContext, useState } from "react";
+import ErrorContext from "ErrorContext";
+
+export function useError(opts) {
+  const [options, setOptions] = useState(opts);
+  const { showError } =  useContext(ErrorContext);
+
+  const triggerError = (params) => {
+      if (options) {
+          if (options.handler) {
+              params.showError = showError;
+              if (params.msg === null) {
+                  params.msg = options.msg;
+              }
+              options.handler(params);
+          } else {
+              if (params) {
+                  params.showError = showError;
+
+                  if (params.handler) {
+                      params.handler(params);
+                      return;
+                  }
+
+                  if (params.error) {
+                      showError(params.error.message);
+                      return;
+                  }
+
+                  if (params.msg) {
+                      showError(params.msg);
+                      return;
+                  }
+
+                  showError(options.msg);
+              } else {
+                  showError(options.msg);
+              }
+          }
+      }
```

Comment:
It is quite difficult to read the flow here, could you invert some of the statements (+ return) to reduce nesting and then maybe also adding a comment that explain what the different scenario we can encounter here.

Reply:
I... made an attempt I think - https://github.com/dfds/selfservice-portal/pull/25/commits/f4f0f43d951a8055836c6b8d8286e7fe1b80f75d 😅

Reply:
wow extremely nice!